service/history/timerGate.go

```diff
@@ -88,7 +88,7 @@ func NewLocalTimerGate() LocalTimerGate {
 	timer := &LocalTimerGateImpl{
 		timer:          time.NewTimer(0),
 		nextWakeupTime: time.Time{},
-		fireChan:       make(chan struct{}),
+		fireChan:       make(chan struct{}, 1),
```

Comment:
Will this cause skipping timer in fireChan?

Reply:
nop, the code, after the change looks like this
```
				select {
				// re-transmit on gateC				
				case timer.fireChan <- struct{}{}:
				default:
				}
```

Reply:
so if there is a signal in the channel already, just skip